pageserver/src/config.rs

```diff
@@ -526,7 +526,6 @@ impl PageServerConf {
 }
 
 #[derive(serde::Deserialize, serde::Serialize)]
-#[serde(deny_unknown_fields)]
```

Comment:
Maybe we should keep this? It seems more like persisted data than config.

Reply:
It's immutable on disk, using `chflags`.

If we ever want to evolve the format, the change will need to be backwards-compatible.